src/utils/cplay.c

```diff
@@ -655,24 +763,55 @@ void play_samples(char *name, unsigned int card, unsigned int device,
 		printf("%s: You should hear audio NOW!!!\n", __func__);
 
 	do {
```

Comment:
This mainly looks good to me but I would avoid  messing the code for suspend/resume with the code for writing frames. 

Can you please implement a function similar to `check_stdin` from alsa-utils?

https://github.com/alsa-project/alsa-utils/blob/master/aplay/aplay.c#L1584